nilearn/datasets/tests/test_func.py

```diff
@@ -362,3 +363,51 @@ def test_fetch_mixed_gambles():
         assert_equal(mgambles["zmaps"][0], os.path.join(datasetdir, "zmaps",
                                                         "sub001_zmaps.nii.gz"))
         assert_equal(len(mgambles["zmaps"]), n_subjects)
+
+
+def test_wrong_inputs_of_dimensionality_timeseriesmethods_matrices():
+    message = ("The %s you have given '%s' is invalid. ")
+    for dim in ['a10', 'd15', 'd30']:
+        assert_raises_regex(ValueError,
+                            message % ('dimensionality', dim),
+                            func.fetch_megatrawls_netmats,
+                            dimensionality=dim)
+
+    for ts in ['ts4', 'st2', 'eigen_regresion', 'mutiple_sptial_regression']:
+        assert_raises_regex(ValueError,
+                            message % ('timeseries', ts),
+                            func.fetch_megatrawls_netmats,
+                            timeseries=ts)
+    for matrices in ['partal_correlation', 'corelation']:
+        assert_raises_regex(ValueError,
+                            message % ('matrices', matrices),
+                            func.fetch_megatrawls_netmats,
+                            matrices=matrices)
+
+
+@with_setup(tst.setup_tmpdata, tst.teardown_tmpdata)
+def test_fetch_megatrawls_netmats():
+    correlations = func.fetch_megatrawls_netmats(data_dir=tst.tmpdir,
```

Comment:
You can take a look at the test for Harvard Oxford fetching in test_atlas.py. Before fetching, we create a dummy XML file otherwise the function crash:
```python
    dummy = open(os.path.join(ho_dir, 'HarvardOxford-Cortical.xml'), 'w')
    dummy.write("<?xml version='1.0' encoding='us-ascii'?> "
                "<metadata>"
                "</metadata>")
    dummy.close()

    ho = atlas.fetch_atlas_harvard_oxford(target_atlas,
                                          data_dir=tst.tmpdir)
```

The problem is that your dummy file must be loadable by `csv_to_array` and empty files are not. You can, for exemple, replace the call of `csv_to_array` by `np.gen_from_txt` directly. You can also determine the filepaths that will be loaded by the function and make them non empty:

```python
    for dim in [25, 100, 200, 300]:
        dummy = open(
            os.path.join(tst.tmp_dir,
                         'Megatrawls',
                         '3T_Q1-Q6related468_MSMsulc_d%d_ts3' % dim , 'Znet2.txt'), 'w')
        dummy.write("1")
        dummy.close()